sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/DirectConnectionConfig.java

```diff
@@ -98,49 +116,57 @@ public DirectConnectionConfig setIdleEndpointTimeout(Duration idleEndpointTimeou
     }
 
     /**
-     * Gets the max channels per endpoint
-     * @return the max channels per endpoint
+     * Gets the max connections per endpoint
+     * Default value is 30
```

Comment:
Am not fully clear on endpointIdletimeout vs connecitonideltimeout. 

Lets please cover at depth the details. 
How about a conceptual MD file in Github which can be referenced?
We can extend it with patterns and guidance (some parts might eventually get promoted to docs). 

Also please refer the performance guide references across code docs extensively.

/cc: @David-Noble-at-work , @moderakh

Reply:
endpoint is a pool of connections per replica. I will improve the documentation around this. 
We initially had a MD file explaining Rntbd architecture. I will work with @David-Noble-at-work  to introduce it back if we can.